crates/store/src/db/mod.rs

```diff
@@ -289,6 +296,25 @@ impl Db {
             })?
     }
 
+    #[instrument(target = "miden-store", skip_all, ret(level = "debug"), err)]
+    pub async fn get_note_sync(
+        &self,
+        block_num: BlockNumber,
+        note_tag_prefixes: &[u32],
+    ) -> Result<NoteSyncUpdate, StateSyncError> {
+        let note_tag_prefixes = note_tag_prefixes.to_vec();
```

Comment:
I've noticed this pattern of taking by reference/slice and then immediately cloning. Bit of an anti-pattern imo

Reply:
Agreed - if a function needs to clone something immediately into a vector, we should just pass it as a vector and let the caller do the cloning, if needed.

Also, we should rename `note_tag_prefixes` into just `note_tags`.